tests/test_dimension.py

```diff
@@ -830,6 +831,98 @@ def test_implicit_dims(self):
 
         assert np.all(f.data == F)
 
+    def test_lowering(self):
+        """
+        Test the lowering of a ConditionalDimension's condition.
+        This test makes an Operator that should indexify and lower the condition
+        passed in the Conditional Dimension
+        """
+
+        grid = Grid(shape=(3, 3))  # Define grid
+
+        # Define Function
+        g = Function(name='g', grid=grid)
+        g.data[:] = 1
+        x, y = grid.dimensions
+        ci = ConditionalDimension(name='ci', parent=y, condition=Le(g, 1))
+
+        f = Function(name='f', shape=grid.shape, dimensions=(x, ci))
+        Operator(Eq(f, g)).apply()
+        assert np.all(f.data == 1)
+
+    def test_expr_like_lowering(self):
```

Comment:
how is this test different than the previous one? maybe just keep this one that uses something slightly more complicated in the condition ie `g1+g2` ?

Reply:
Yes